src/libstd/sys/unix/thread.rs

```diff
@@ -180,7 +182,7 @@ pub mod guard {
 }
 
 
-#[cfg(any(target_os = "linux",
+#[cfg(any(all(target_os = "linux", not(target_env = "musl")),
```

Comment:
I haven't looked too deeply, but I was getting crashes in one of the pthread calls in this code path, and since it's some hairy code I just punted.